sources/light_client.move

```diff
@@ -57,7 +56,7 @@ fun init(_ctx: &mut TxContext) {
 ///
 /// Encode header reference:
 /// https://developer.bitcoin.org/reference/block_chain.html#block-headers
-public(package) fun new_light_client_with_params(params: Params, start_height: u64, trusted_headers: vector<vector<u8>>, start_chain_work: u256, ctx: &mut TxContext): LightClient {
+public fun new_light_client_with_params(params: Params, start_height: u64, trusted_headers: vector<vector<u8>>, start_chain_work: u256, ctx: &mut TxContext): LightClient {
```

Comment:
I still refer to the public(package) here because we have new_light_client for creating the lc object.

Reply:
the idea is to be able to setup custom params. What's the reason to not allow for this?

Reply:
We allow this in new_light_client. the parameters we need to use always in regtest, mainnet, and testnet. Additionally,  we don't transfer or make this share, so in this case, we will receive an error when creating an LC object using this function.

Reply:
I mean, I receive an error when calling this function outside the tests module.

Reply:
Why? I think people should be able to use it as they want. They may use different Paramus for their private testnets

Reply:
Yes, if you do this, please ensure we make the LC object "share."  I recommend creating another function but wrapping it with the new_light_client_with_params function to avoid breaking changes.

Reply:
Good catch. Updated, and moved event and transfer::share to the new_light_client_with_params